src/wallet.h

```diff
@@ -304,9 +304,9 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
         fBackupMints = false;
 
         // Stake Settings
-        nHashDrift = 45;
+        nHashDrift = 60;
         nStakeSplitThreshold = 2000;
-        nHashInterval = 22;
+        nHashInterval = 2;
```

Comment:
This means that every 2 seconds you will be hashing the next 60 seconds of timestamps on each UTXO that you have.
So the first round of hashing would do seconds 0-60 for each UTXO, the next round of hashing would do seconds 2-62 for each UTXO. That creates 58 redundant hashes per UTXO held for each round of hashing.

Right now the settings are at 22/30 because that seemed to be a decent balance between having some overlapping hashes as well as not hashing too far out into the future.

Hashing out into the future can be more likely to create an orphan stake. If two people get a valid stake at the same time, whoever has the stake with the lowest timestamp will win the block. The second person is better off waiting until the time is closer and then publishing the stake when there is less time for someone else to produce a stake that orphans it.

Reply:
The frequency of hash searches is intended. (Every 2 seconds)
Duplicate nTimeTx(hash) searches only occur to consume more CPU time for testing. I agree that it is the part that should be optimized.

Currently, orphans occur more frequently due to longer interval + long zpiv spending time.
(Even if it is a long interval, I want to reduce it for now.)

Do not change the nHashDrift.  I hope it works as follows.

before : 0-30 scan > sleep 22 sec > 31-60 scan
after : 0-30 scan > sleep 1 sec > 1-31 scan > sleep 1 sec > 2-32 scan

It is right to eliminate the duplication.

There is one more important thing.

In the code example below, nTimeBlockFrom and nStakeModifier are the UTXO fixed values. That is correct for CPivStake. However, it is no longer fixed in the CZPivStake.
Therefore, the next round of navigation requires a full hash search and short interval.

```

for (std::unique_ptr<CStakeInput>& stakeInput : listInputs)
{
    CBlockIndex* pindex = stakeInput->GetIndexFrom();
    CBlockHeader block = pindex->GetBlockHeader();

    unsigned int nTimeBlockFrom;

    nTimeBlockFrom = block.GetBlockTime();

    uint64_t nStakeModifier = 0;
    stakeInput->GetModifier(nStakeModifier); 
}
```

Reply:
`nTimeBlockFrom` is not used as a hash input, only to double check that the input being staked is old enough to be staked. This is already checked in `SelectStakeCoins()` so should not be a huge concern as long as the stake set is updated every few minutes with the current available coins to stake.

The `nStakeModifier` for the zpiv stake changes less often since it is based on the first accumulator checkpoint that occurs 60 minutes after `pindexFrom`. Accumulator checkpoints change maximum of every 10 blocks, sometimes it can be an even longer span than that too. So this part of the hash is redundant a lot of the time, but not always.